### PR TITLE
Update SPL class construtor pages

### PR DESCRIPTION
**Pull request by Girgias**
QA: remove return value section

Update some error sections

I dropped the error section when the exception was a ZPP like exception.

### CODE REVIEW

File: reference/spl/globiterator/construct.xml [hunk 1]
```diff
@@ -43,6 +43,19 @@
   </para>
  </refsect1>
 
+ <refsect1 role="errors">
+  &reftitle.errors;
+  <para>
+   Throws an <classname>UnexpectedValueException</classname>
+   if the <parameter>directory</parameter> cannot be opened.
```
Comment (cmb69) [inline reference/spl/globiterator/construct.xml:50]: "does not exist"?

File: reference/spl/limititerator/construct.xml [hunk 2]
```diff
@@ -53,19 +53,13 @@
...
- </refsect1>
-
  <refsect1 role="errors">
   &reftitle.errors;
   <para>
-   Throws an <classname>OutOfRangeException</classname>
+   Throws a <classname>ValueError</classname>
    if the <parameter>offset</parameter> is less than <literal>0</literal>
    or the <parameter>limit</parameter> is less than <literal>-1</literal>.
+   Prior to PHP 8.0.0, it threw an <classname>OutOfRangeException</classname>.
```
Comment (cmb69) [inline reference/spl/limititerator/construct.xml:62]: Changelog?

File: reference/spl/recursivedirectoryiterator/construct.xml [hunk 3]
```diff
@@ -46,18 +46,16 @@
...
-   Returns the newly created <classname>RecursiveDirectoryIterator</classname>.
-  </para>
- </refsect1>
-
  <refsect1 role="errors">
   &reftitle.errors;
   <para>
    Throws an <classname>UnexpectedValueException</classname>
-   if the <parameter>directory</parameter> cannot be found or is not a directory.
+   if the <parameter>directory</parameter> cannot be opened.
```
Comment (cmb69) [inline reference/spl/recursivedirectoryiterator/construct.xml:53]: Wording (see above)

File: reference/spl/recursivedirectoryiterator/construct.xml [hunk 4]
```diff
@@ -46,18 +46,16 @@
...
+  </para>
+  <para>
+   Throws a <classname>ValueError</classname>
+   if the <parameter>directory</parameter> is an empty string.
+   Prior to PHP 8.0.0, it threw a <classname>RuntimeException</classname>.
```
Comment (cmb69) [inline reference/spl/recursivedirectoryiterator/construct.xml:58]: Changelog?

File: reference/spl/splfixedarray/construct.xml [hunk 5]
```diff
@@ -33,22 +33,13 @@
...
   &reftitle.errors;
   <para>
-   Throws <classname>InvalidArgumentException</classname> when <parameter>size</parameter> is a negative
-   number.
-  </para>
-  <para>
-   Raises <constant>E_WARNING</constant> when <parameter>size</parameter> cannot be parsed as a number.
+   Throws a <classname>ValueError</classname> when <parameter>size</parameter>
+   is a negative integer.
+   Prior to PHP 8.0.0, it threw a <classname>InvalidArgumentException</classname>.
```
Comment (cmb69) [inline reference/spl/splfixedarray/construct.xml:42]: Changelog?

File: reference/spl/arrayobject/construct.xml [hunk 6]
```diff
@@ -53,29 +53,11 @@
...
-     </simpara>
-    </listitem>
-    <listitem>
-     <simpara>
-      <parameter>flags</parameter> is not an integer
-     </simpara>
-    </listitem>
     <listitem>
      <simpara>
       <parameter>iteratorClass</parameter> is not an object that implements
```
[the remark this answers is not in the document]
Comment (Girgias) [reply reference/spl/arrayobject/construct.xml:63]: No, ZPP only checks that the argument is a valid class, the check for Iterator seems to be here: https://heap.space/xref/php-src/ext/spl/spl_array.c?r=db0db220#1026

[the remark this answers is not in the document]
Comment (cmb69) [reply reference/spl/arrayobject/construct.xml:63]: That code is secondary in this case; as of PHP 7.0.0, [ZPP checks for "instance of iterator"](https://github.com/php/php-src/blob/PHP-7.0/ext/spl/spl_array.c#L1203), and a [`TypeError` is thrown](https://3v4l.org/OuLlf) if that fails (no longer the `InvalidArgumentException` as documented).

**Review by cmb69 — APPROVED** [review]

**Pull request merged**